src/main/scala/com/databricks/labs/overwatch/validation/SnapValidation.scala

```diff
@@ -127,46 +135,76 @@ class SnapValidation(params: SnapValidationParams,
    * @param params
    * @return
    */
-  protected def snapTable(target: PipelineTable): SnapReport = {
-    val databaseTable = s"${params.snapDatabaseName}.${target.name}"
-    val tableName = s"${params.sourceDatabaseName}.${target.name}"
+  protected def snapTable(bronzeModule: Module, target: PipelineTable): SnapReport = {
+
+    val module = bronzeModule.copy(_moduleDependencies = Array[Int]())
+    val pipeline = module.pipeline
 
     try {
-      getFilteredDF(target, tableName, startSnap, endCompare)
-        .repartition()
-        .write
-        .format("delta")
-        .mode("overwrite")
-        .option("overwriteSchema", "true")
-        .saveAsTable(databaseTable)
+      val finalDFToClone = target.copy(_databaseName = sourceDB).asIncrementalDF(module, target.incrementalColumns)
+      pipeline.database.write(finalDFToClone, target, pipeline.pipelineSnapTime.asColumnTS)
+
 
       println(s"SNAPPED: ${target.tableFullName}")
+      val snapCount = spark.table(target.tableFullName).count()
+
+      println(s"UPDATING Module State")
+      pipeline.updateModuleState(module.moduleState.copy(
```

Comment:
update pipeline state as it's done in the code so that subsequent runs will be able to progress the pipeline state.